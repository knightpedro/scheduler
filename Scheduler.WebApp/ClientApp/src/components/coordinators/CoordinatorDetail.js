import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import {
  coordinatorsSelectors,
  deleteCoordinator,
} from "../../ducks/coordinators";
import { Empty } from "../common";
import { Grid, Header, Icon, Button, Modal, Divider } from "semantic-ui-react";
import { useHistory } from "react-router-dom";
import routes from "../../routes";
import { CoordinatorFormContainer } from "../forms/containers";
import CoordinatorJobsTable from "./CoordinatorJobsTable";

const CoordinatorDetail = ({ id }) => {
  const dispatch = useDispatch();
  const history = useHistory();
  const [showForm, setShowForm] = useState(false);
  const [showModal, setShowModal] = useState(false);
  const coordinator = useSelector((state) =>
    coordinatorsSelectors.selectById(state, id)
  );

  const handleDelete = () => {
    dispatch(deleteCoordinator(id)).then(() => {
      history.push(routes.coordinators.base);
    });
  };

  if (!coordinator) return <Empty basic message="Coordinator not found" />;

  return (
    <>
      <Grid>
        <Grid.Row columns="equal">
          <Grid.Column>
            <Header as="h2">
              <Icon
                name="user"
                color={coordinator.isActive ? "green" : "red"}
              />
              <Header.Content>
                {coordinator.name}
                <Header.Subheader>{coordinator.email}</Header.Subheader>
              </Header.Content>
            </Header>
          </Grid.Column>
          <Grid.Column textAlign="right">
            <Button.Group>
              <Button icon="edit" onClick={() => setShowForm(true)} />
              <Button icon="trash" onClick={() => setShowModal(true)} />
            </Button.Group>
          </Grid.Column>
        </Grid.Row>
        {showForm && (
          <Grid.Row columns="equal">
            <Grid.Column>
              <CoordinatorFormContainer
                id={id}
                closeForm={() => setShowForm(false)}
                showHeader={false}
              />
            </Grid.Column>
          </Grid.Row>
        )}
        <Divider hidden />
        <Grid.Row columns="equal">
          <Grid.Column>
            <Header>Jobs</Header>
            <CoordinatorJobsTable id={id} />
          </Grid.Column>
        </Grid.Row>
      </Grid>
      {showModal && (
        <Modal open={showModal} onClose={() => setShowModal(false)}>
          <Header content="Delete Coordinator" />
          <Modal.Content>
            Are you sure you want to delete {coordinator.name}?
          </Modal.Content>
          <Modal.Actions>
            <Button onClick={() => setShowModal(false)} content="Cancel" />
            <Button onClick={handleDelete} content="Delete" negative />
          </Modal.Actions>
        </Modal>
      )}
    </>
  );
};

export default CoordinatorDetail;
