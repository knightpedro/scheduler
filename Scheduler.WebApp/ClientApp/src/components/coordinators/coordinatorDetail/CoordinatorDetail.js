import React, { useState } from "react";
import { EditDeleteGroup } from "../../common/actions";
import Routes from "../../../routes";
import { generatePath } from "react-router-dom";
import { CentredModal } from "../../common/modals";
import JobsList from "../../jobs/jobsList/JobsList";
import { ActiveStatus } from "../../common/status";
import NoContent from "../../common/noContent";

const DELETE_MESSAGE = "Are you sure you want to delete this coordinator?";

const CoordinatorDetail = ({ coordinator, handleDelete }) => {
  const [showModal, setShowModal] = useState(false);
  const editPath = generatePath(Routes.coordinators.EDIT, {
    id: coordinator.id
  });

  const handleModalSuccess = () => {
    setShowModal(false);
    handleDelete();
  };

  return (
    <>
      <div className="row align-items-center">
        <div className="col-auto">
          <h2>{coordinator.name}</h2>
        </div>
        <div className="col mr-auto">
          <ActiveStatus isActive={coordinator.isActive} />
        </div>
      </div>

      <div className="row align-items-center">
        <div className="col mr-auto"></div>
        <div className="col-auto">
          <EditDeleteGroup
            editPath={editPath}
            handleDeleteClick={() => setShowModal(true)}
          />
        </div>
      </div>

      {coordinator.jobs.length > 0 ? (
        <JobsList jobs={coordinator.jobs} />
      ) : (
        <NoContent item="jobs" />
      )}
      <CentredModal
        show={showModal}
        title={`Delete ${coordinator.name}`}
        content={DELETE_MESSAGE}
        handleSuccess={handleModalSuccess}
        handleHide={() => setShowModal(false)}
      />
    </>
  );
};

export default CoordinatorDetail;
