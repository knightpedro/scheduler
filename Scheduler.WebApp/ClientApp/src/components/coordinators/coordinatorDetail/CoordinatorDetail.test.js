import React from "react";
import { shallow } from "enzyme";
import CoordinatorDetail from "./CoordinatorDetail";

const ActiveCoordinator = {
    id: 1,
    isActive: true,
    name: "Active Coordinator",
    jobs: [],
};

const InactiveCoordinator = {
    id: 1,
    isActive: false,
    name: "Inactive Coordinator",
    jobs: [],
};

describe("Coordinator Detail", () => {
    it("renders correctly", () => {
        const wrapper = shallow(
            <CoordinatorDetail coordinator={ActiveCoordinator} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("renders NoContent when there are no jobs", () => {
        const wrapper = shallow(
            <CoordinatorDetail coordinator={ActiveCoordinator} />
        );
        expect(wrapper.find("NoContent")).toHaveLength(1);
    });
});
