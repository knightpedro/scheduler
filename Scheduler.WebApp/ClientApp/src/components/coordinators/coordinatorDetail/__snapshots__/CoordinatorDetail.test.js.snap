// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coordinator Detail renders correctly 1`] = `
<Fragment>
  <div
    className="row align-items-center"
  >
    <div
      className="col-auto"
    >
      <h2>
        Active Coordinator
      </h2>
    </div>
    <div
      className="col mr-auto"
    >
      <ActiveStatus
        isActive={true}
        showLabel={true}
      />
    </div>
  </div>
  <div
    className="row align-items-center"
  >
    <div
      className="col mr-auto"
    />
    <div
      className="col-auto"
    >
      <EditDeleteGroup
        editPath="/coordinators/edit/1"
        handleDeleteClick={[Function]}
      />
    </div>
  </div>
  <NoContent
    item="jobs"
  />
  <CentredModal
    content="Are you sure you want to delete this coordinator?"
    handleHide={[Function]}
    handleSuccess={[Function]}
    show={false}
    title="Delete Active Coordinator"
  />
</Fragment>
`;
